pegen/c_generator.py

```diff
@@ -87,21 +87,30 @@ class CCallMakerVisitor(GrammarVisitor):
     def __init__(self, parser_generator: ParserGenerator):
         self.gen = parser_generator
         self.cache: Dict[Any, Any] = {}
+        self.keyword_cache: Dict[str, int] = {}
+
+    def keyword_helper(self, keyword: str) -> Tuple[str, str]:
+        if keyword not in self.keyword_cache:
+            self.keyword_cache[keyword] = self.gen.keyword_type()
+        return "keyword", f"expect_token(p, {self.keyword_cache[keyword]})"
 
     def visit_NameLeaf(self, node: NameLeaf) -> Tuple[str, str]:
         name = node.value
         if name in ("NAME", "NUMBER", "STRING"):
             name = name.lower()
             return f"{name}_var", f"{name}_token(p)"
-        if name in ("NEWLINE", "DEDENT", "INDENT", "ENDMARKER", "ASYNC", "AWAIT"):
+        if name in ("NEWLINE", "DEDENT", "INDENT", "ENDMARKER"):
             name = name.lower()
             return f"{name}_var", f"{name}_token(p)"
+        if name in ("ASYNC", "AWAIT"):
+            name = name.lower()
+            return self.keyword_helper(name)
```

Comment:
But `ASYNC` and `AWAIT` are tokens returned by the tokenizer -- they don't need the keyword treatment here. I think you can just keep the old code for these.